odoo/tools/misc.py

```diff
@@ -1184,18 +1184,21 @@ def clear(self):
         self.data.clear()
 
 
-class IterableGenerator:
-    """ An iterable object based on a generator function, which is called each
-        time the object is iterated over.
-    """
-    __slots__ = ['func', 'args']
-
-    def __init__(self, func, *args):
-        self.func = func
-        self.args = args
+class IterableGenerator(ABC):
```

Comment:
Not only has the name become wholly unsuitable (as the semantics have completely changed and there is no generator function being wrapped / called anymore), this class seems rather unnecessary at that point, it's just converting forward/backward iteration to a boolean which the implementer has then translate back to forwards/backwards iteration.

Doesn't seem like it'd be a huge chore for the implementers to do that on their own, with an internal helper in the odd case where it really makes sense.